### ISSUE TITLE
SSH vagrant default password is not working. Here are steps to reproduce.

### ISSUE THREAD

**Issue by Jakobud** [puphpet/puphpet#2357]
Yes, I know I can just type `# vagrant ssh` to ssh into my VM. But I'm attempting to use SSH as a tunnel to connect to the VM for database manipulation and some other things. So I need to actually ssh using the vagrant username and password.

According to documentation, the default username/password is `vagrant/vagrant`. But that password is not working. Here are the exact steps to reproduce the problem and prove the password was not set:

### Step 1: Try to ssh in using `vagrant/vagrant`
```
# ssh vagrant@127.0.0.1 -p 2222
vagrant@127.0.0.1's password: 
Permission denied, please try again.
vagrant@127.0.0.1's password: 
```
So that doesn't work. `^c` that.

### Step 2: ssh in using vagrant method
```
jake@macbook:~/vm/webdev01
# vagrant ssh
Welcome to Ubuntu 14.04.3 LTS (GNU/Linux 3.19.0-25-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
----------------------------------------------------------------
  Ubuntu 14.04.3 LTS                          built 2015-12-01
----------------------------------------------------------------
Last login: Fri Jun 24 16:21:40 2016 from 10.0.2.2
```

### Step 3: Set the user's password to `vagrant`
```
[04:22 PM]-[vagrant@webdev01]-[~] 
$ passwd
Enter new UNIX password: 
Retype new UNIX password: 
passwd: password updated successfully
```

### Step 4: Exit the ssh session, back to the host
```
[04:22 PM]-[vagrant@webdev01]-[~] 
$ exit
logout
Connection to 127.0.0.1 closed.
```

### Step 5: ssh again, using `vagrant/vagrant`
```
jake@macbook:~/vm/webdev01
# ssh vagrant@127.0.0.1 -p 2222
vagrant@127.0.0.1's password: 
Welcome to Ubuntu 14.04.3 LTS (GNU/Linux 3.19.0-25-generic x86_64)

 * Documentation:  https://help.ubuntu.com/
----------------------------------------------------------------
  Ubuntu 14.04.3 LTS                          built 2015-12-01
----------------------------------------------------------------
Last login: Fri Jun 24 16:22:10 2016 from 10.0.2.2

[04:22 PM]-[vagrant@webdev01]-[~] 
$ 
```

### SUCCESS

So you can clearly see, that with PuPHPet, the default vagrant password is not working. I have no idea what it is set to. I tried it blank, `123`, `password`, etc. It wasn't until I went in and explicitly set the password that it worked.

So what is going on here? Where is the problem? How can I fix this so that the password is properly set during provisioning? Or is this a PuPHPet bug?

Here is my `ssh-config` and `config.yaml` for reference.

```
# vagrant ssh-config
Host webdev01
  HostName 127.0.0.1
  User vagrant
  Port 2222
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no
  PasswordAuthentication no
  IdentityFile /Users/jake.wilson/vm/webdev01/puphpet/files/dot/ssh/id_rsa
  IdentityFile /Users/jake.wilson/.vagrant.d/insecure_private_key
  IdentitiesOnly yes
  LogLevel FATAL
```

My `config.yaml`:
```
vagrantfile:
    target: local
    vm:
        provider:
            local:
                box: puphpet/ubuntu1404-x64
                box_url: puphpet/ubuntu1404-x64
                box_version: '0'
                chosen_virtualizer: virtualbox
                virtualizers:
                    virtualbox:
                        modifyvm:
                            natdnshostresolver1: false
                        showgui: 0
                    vmware:
                        numvcpus: 1
                    parallels:
                        use_linked_clone: 0
                        check_guest_tools: 0
                        update_guest_tools: 0
                machines:
                    vflm_rdcrvl133kgk:
                        id: webdev01
                        hostname: webdev01.dev
                        network:
                            private_network: 80.80.80.80
                            forwarded_port:
                                vflmnfp_invj0ssna772:
                                    host: '5367'
                                    guest: '22'
                        memory: '512'
                        cpus: '1'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder:
            vflsf_kzr53rzw3vze:
                source: ./
                target: /vagrant
                sync_type: default
                smb:
                    smb_host: ''
                    smb_username: ''
                    smb_password: ''
                    mount_options:
                        dir_mode: '0775'
                        file_mode: '0664'
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
        post_up_message: ''
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        password: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
server:
    install: '1'
    packages:
        - vim
        - htop
        - wget
        - curl
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: en_US.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: UTC
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '1'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_dyo0d4u0g1s8:
            server_name: mywebsite.dev
            server_aliases:
                - mywebsite.dev
            www_root: /vagrant/vhosts/mywebsite.dev/public
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_pczzjns0desg:
                    location: /
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_sk3b61ae475d:
                    location: '~ \.php$'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '0'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_ec9ghslv6jtk:
            servername: awesome.dev
            serveraliases:
                - www.awesome.dev
            docroot: /var/www/awesome
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_kdb2zdfpvqx0:
                    path: /var/www/awesome
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_ewqj59v2mdmz:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
letsencrypt:
    install: '1'
    settings:
        email: admin@vagrant.dev
    domains: {  }
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
            - curl
            - mcrypt
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_bvoue919d3tv:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: www-user
                group: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.21.1
drush:
    install: '0'
    version: 8.0.5
ruby:
    install: '1'
    versions:
        rv_4pfz0f778osg:
            default: '1'
            bundler: '1'
            version: 1.9.3
            gems:
                - deep_merge
                - activesupport
                - vine
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '1'
    settings:
        version: '5'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '1'
    settings:
        version: '10.1'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_3gp1a7njvyqf:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_dr9ymrt7awh5:
            name: dbname
            sql: ''
    grants:
        mariadbng_feldnqsaobqy:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '0'
    settings:
        version: '5.6'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_vy69szgqycza:
            name: dbuser
            password: '123'
    databases:
        mysqlnd_f8chh7jhim5r:
            name: dbname
            sql: ''
    grants:
        mysqlng_5bsrlpqgcwnv:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.3.1
        java_install: true
    instances:
        esi_2102orvn0ttx:
            name: es-01
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```

**Comment by exchentrix** [comment]
There is no default password for SSHing into your machine. Instead of that, you should use your RSA key pair which is located in "puphpet/files/dot/ssh"

**Comment by Jakobud** [comment]
And how would I use an RSA key pair to SSH tunnel using something like Sequel Pro on OSX?

**Comment by exchentrix** [comment]
I'm not familiar with Sequel Pro, I use MySQL-Workbench. But it seems, that Sequel Pro also support key pair authentication: http://www.sequelpro.com/docs/Connecting_to_a_MySQL_Server_on_a_Remote_Host

**Comment by exchentrix** [comment]
test


Am 26.06.2016 um 09:49 schrieb Jake Wilson:

**Comment by Jakobud** [comment]
Okay so where is my the location of my key pair file I need to use here? Here is my ssh Puppet config:

```
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        password: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
```

I see no files in `~/.ssh/` or in my host directory. I'm obviously able to ssh to my guest machine using some sort of key pair authentication but I have no idea where this file is.

**Comment by exchentrix** [comment]
The ssh-key is generated when you vagrant up your machine. It is not located in your home directory under ".ssh". Take a look at "puphpet/files/dot/ssh", the files should be there.

**Comment by Jakobud** [comment]
Ah I found it. By default Vagrant uses an insecure private key for all instances here `~/.vagrant.d/insecure_private_key`. It works! Thank you for your insight.

**Comment by jamfwww** [comment]
I came into work today and suddenly Sequel Pro would not connect with the same settings I've had in place for a year. No other environment changes. I renamed the private_key file (.vagrant/machines/[name]/virtualbox/private_key) so Vagrant wouldn't see it, then ran vagrant ssh-config, which showed the new IdentifyFile as /Users/[me]/.vagrant.d/insecure_private_key (as @Jakobud pointed out). I changed Sequel Pro to use that key and it started working again.

**Comment by litzinger** [comment]
I came into work today and suddenly Sequel Pro would not connect with the same settings I've had in place for a year. No other environment changes. I renamed the private_key file (.vagrant/machines/[name]/virtualbox/private_key) so Vagrant wouldn't see it, then ran vagrant ssh-config, which showed the new IdentifyFile as /Users/[me]/.vagrant.d/insecure_private_key (as @Jakobud pointed out). I changed Sequel Pro to use that key and it started working again.
